### ISSUE TITLE
Support the <dialog> element

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3463]
Support for the dialog element has picked up quite a bit in the last year. We need to make sure that while closed, axe-core doesn't flag content inside a dialog element, and when opened, that axe-core properly considers the implications of things like contrast on the background, and that elements in the background are not focusable.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog

**Comment by straker** [comment]
Some investigation into support:

- A closed dialog has browser styling of `display: none` so anything inside a close dialog is treated as not visible (i.e. no rules are run)
- An opened dialog overrides the display and uses `display: block` so all rules are run as normal
- A dialog opened by adding the `open` attribute in all browsers has no backdrop and does not prevent elements outside of the dialog from being focused or interacted with

Now here is where things get tricky.

- A modal opened using the `showModal` method [blocks the document element](https://html.spec.whatwg.org/multipage/interaction.html#blocked-by-a-modal-dialog). This means elements outside the dialog are not focusable or intractable 
- A modal opened using the `showModal` method adds a transparent backdrop to the dialog

As far as I can tell, there is no way to determine when a dialog is opened via `showModal` or just by adding the `open` attribute. Theoretically each modal has an ["is modal" flag](https://html.spec.whatwg.org/multipage/interactive-elements.html#is-modal) which would allow us to determine if it's a modal or not, but it doesn't look like this flag is accessible from JavaScript.

Similarly, since there's no way to tell when a dialog is a modal, there's nothing to determine when  all elements outside the modal should be treated as inert. `inert` is not added to the page or on any element and instead each element is treated intrinsically as `inert` by the browser.

A modal that is opened using `showModal` has [different browser styling](https://html.spec.whatwg.org/multipage/rendering.html#flow-content-3:is-modal) than one that just uses the `open` attribute (such as `position: fixed`), but I don't think that would be a reliable indicator of a modal dialog since users can override it using CSS. I also tried looking at the styles of the `::backdrop` pseudo element, but trying to do that still returns a style even if the dialog does not have a backdrop.

**Comment by padmavemulapati** [comment]
Validated with the latest code base, 
with the dialog and showModel(), it is not failing with the below code snippet:
```
<dialog open>
    <p>Greetings, one and all!</p>
    <form method="dialog">
      <button>OK</button>
    </form>
  </dialog>
  <!-- A modal dialog containing a form -->
<dialog id="favDialog">
  <form method="dialog">
    <p>
      <label>Favorite animal:
        <select>
          <option value="default">Choose…</option>
          <option>Brine shrimp</option>
          <option>Red panda</option>
          <option>Spider monkey</option>
        </select>
      </label>
    </p>
    <div>
      <button value="cancel">Cancel</button>
      <button id="confirmBtn" value="default">Confirm</button>
    </div>
  </form>
</dialog>
<p>
  <button id="showDialog">Show the dialog</button>
</p>
<output></output>
<script>const showButton = document.getElementById('showDialog');
  const favDialog = document.getElementById('favDialog');
  const outputBox = document.querySelector('output');
  const selectEl = favDialog.querySelector('select');
  const confirmBtn = favDialog.querySelector('#confirmBtn');
  
  // "Show the dialog" button opens the <dialog> modally
  showButton.addEventListener('click', () => {
      favDialog.showModal();
  });
  // "Favorite animal" input sets the value of the submit button
  selectEl.addEventListener('change', (e) => {
    confirmBtn.value = selectEl.value;
  });
  // "Confirm" button of form triggers "close" on dialog because of [method="dialog"]
  favDialog.addEventListener('close', () => {
    outputBox.value = `ReturnValue: ${favDialog.returnValue}.`;
  });</script>
```
Before selecting the dialog:


![Image](https://user-images.githubusercontent.com/47409579/229017571-03bcb6df-e230-4196-ad1c-c4e679aa7ea6.png)

When it retuns the show item:


![Image](https://user-images.githubusercontent.com/47409579/229017645-fb5f3b61-53fb-4608-8e35-2dd3a1009765.png)
